src/ppl/src/SimulatedAnnealing.cpp

```diff
@@ -185,9 +187,11 @@ int SimulatedAnnealing::randomAssignmentForGroups(
       pin_assignment_[pin_idx] = group_slot;
       slots_[group_slot].used = true;
       group_slot++;
-      slot_idx++;
       placed_pins.insert(pin_idx);
     }
+    while (slots_[slot_idx].used) {
+      slot_idx++;
+    }
```

Comment:
This guarantees that you start on an empty slot but I'm not clear it prevents overlap (you might start the next group too close to the next group to fit).